utils/dataset_manifest/core.py

```diff
@@ -195,8 +195,11 @@ def __iter__(self):
                 image = next(sources)
                 img = Image.open(image, mode='r')
                 orientation = img.getexif().get(274, 1)
-                img_name = os.path.relpath(image, self._data_dir) if self._data_dir \
-                    else os.path.basename(image)
+                if isinstance(image, str):
+                    img_name = os.path.relpath(image, self._data_dir) if self._data_dir \
+                        else os.path.basename(image)
+                else:
```

Comment:
Which type this value is supposed to have?

Reply:
BytesIO, but not relevant anymore